backend/user/DbHandler.cs

```diff
@@ -0,0 +1,443 @@
+            object name = await cmd.ExecuteScalarAsync();
+            return name == null ? "" : (string) name;
+        }
+        public async Task<int> addGame(int p1, int p2, JObject board) {
+            try {
+                await using var dbCon = await DBConnection.NewConnectionAsync();
+                //add a game row in Game table
+                Console.WriteLine($"Adding a row in game table for ({p1},{p2}");
+                string insert_query = "insert into game (p1,p2,board) values (@p1, @p2, @board)";
+                await using var insert_cmd = new MySqlCommand(insert_query, dbCon);
+                insert_cmd.Parameters.AddWithValue("@p1", p1);
+                insert_cmd.Parameters.AddWithValue("@p2", p2);
+                insert_cmd.Parameters.AddWithValue("@board", board);
+                await insert_cmd.ExecuteNonQueryAsync();
+                long id = insert_cmd.LastInsertedId;
+                return (int) id;
+
+            } catch (Exception e) {
+                Console.WriteLine($"Cannot connect to db: {e.Message}");
+                throw;
+            }
+        }
+
+        public async Task markUserAvailable(int uid) {
+            try {
+                await using var dbCon = await DBConnection.NewConnectionAsync();
+                //add a game row in Game table
+                Console.WriteLine($"Makr user available{uid}");
+                string query = "update user set want_play=true where uid=@uid";
+                await using var insert_cmd = new MySqlCommand(query, dbCon);
+                insert_cmd.Parameters.AddWithValue("@uid", uid);
+                await insert_cmd.ExecuteNonQueryAsync();
+
+            } catch (Exception e) {
+                Console.WriteLine($"Cannot connect to db: {e.Message}");
+                throw;
+            }
+        }
+
+        public async Task<int> getGidForUser(int uid) {
+            try {
+                await using var dbCon = await DBConnection.NewConnectionAsync();
+                //add a game row in Game table
+                Console.WriteLine($"Makr user available{uid}");
+                string query = "select gid from user where uid=@uid";
+                await using var insert_cmd = new MySqlCommand(query, dbCon);
+                insert_cmd.Parameters.AddWithValue("@uid", uid);
+                var id = await insert_cmd.ExecuteScalarAsync();
+                return (int) id;
+
+            } catch (Exception e) {
+                Console.WriteLine($"Cannot connect to db: {e.Message}");
+                throw;
+            }
+        }
+
+        public async Task<int> createGameAsync(int p1) {
+            try {
+                await using var dbCon = await DBConnection.NewConnectionAsync();
+
+                MySqlCommand myCommand = dbCon.CreateCommand();
+                MySqlTransaction myTrans;
+
+                // Start a local transaction
+                myTrans = dbCon.BeginTransaction();
```

Comment:
Change these two lines to

(maybe await) using var myTrans = await dbCon.BeginTransactionAsync();